### ISSUE TITLE
Use only a single TransactionStatus enum

### ISSUE THREAD

**Issue by milancermak** [xJonathanLEI/starknet-rs#9]
We currently have `TransactionStatus` and `TransactionStatusType` which are *almost* the same. The only difference is that the former also wraps a block hash with two of its variants. `TransactionStatus` is used only in a single place, as the response type for `get_transaction_status`.

In my opinion, this is quite confusing. Both types do pretty much the same job. I'm suggesting to introduce a different type to replace the current `TransactionStatus`, something like:

```rust
pub struct ShortTransactionStatus { // name is of course open for discussion
    pub block_hash: Option<H256>,
    pub tx_status: TransactionStatusType
}
```

It would be used as the return type for `get_transaction_status`. That way, we could remove `TransactionStatus` and also `TransactionBlockHash`, because that would just be "merged" into this new struct and have a cleaner API.

**Comment by milancermak** [comment]
As I was writing the tests and digging through the types deeper, I came across `TransactionWithStatus` which could be refactored in sync with the proposed changes. An example (I don't have it fully fleshed out yet):

```rust
pub enum Transaction {
    pub Short(ShortTransaction)
    pub Full(FullTransaction)
}

pub struct ShortTransaction {
    pub block_hash: Option<H256>,
    pub tx_status: TransactionStatusType
}

pub struct FullTransaction { // just a new name for the current TransactionWithStatus
    // ...
}
```

We could probably use the `Transaction` for both `/get_transaction` and `/get_transaction_status` calls.

Does it make sense to you @xJonathanLEI?

**Comment by xJonathanLEI** [comment]
Hmm you're right that the types around transactions are a bit messy, thanks to the inconsistency of the API schema...

I'm not making a stance here, but for the sake of explaining the original design decision, I was wrapping the block hash inside an enum variant so that users can directly extract the block hash idiomatically inside a `match` arm, for example, without calling `unwrap()`.

**Comment by xJonathanLEI** [comment]
If we decide to diverge from gateway response shape, we'll need to move the deserialization tests from `core` to `providers`, since they'll no longer be the same representation.

**Comment by milancermak** [comment]
Yeah, I understand the original design decision, the API isn't the cleanest to work with.

By giving up the idiomatic handling with `match` we gain much more understandable types. I think it's a worthwhile tradeoff to make. Users of the library will be accustomed to calling `unwrap` since it's a common occurrence in Rust, but will have to learn the types it exposes and wonder what's the difference between `TransactionStatus` and `TransactionStatusType`.

**Comment by xJonathanLEI** [comment]
My point on `unwrap()` was more on idiomaticness, since `unwrap()` potentially panics. It's more of a peace of mind thing. But I agree that this is a tradeoff.

**Comment by milancermak** [comment]
Resolved in #60.
